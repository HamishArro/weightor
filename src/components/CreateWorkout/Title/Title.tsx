import React from 'react';
import {AntDesign} from '@expo/vector-icons';
import {Text, View} from 'react-native';
import {styles} from '../../../utils/styleSheet';
import {TitleProps} from '../types';

function Title({title, handleBack}: TitleProps) {
  return (
    <View style={styles.titleContainer} testID={'title-container'}>
      <Text children={title} style={styles.title} />
      <AntDesign
        name="back"
        testID="back-button"
        onPress={handleBack}
        size={20}
        color="white"
      />
    </View>
  );
}

export default Title;
