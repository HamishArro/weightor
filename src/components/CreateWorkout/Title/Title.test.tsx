import * as React from 'react';
import {describe, it, beforeEach, expect, jest} from '@jest/globals';
import {fireEvent, render, screen} from '@testing-library/react-native';
import Title from './Title';

const props = {title: 'title', handleBack: jest.fn()};

describe('Title tests', () => {
  beforeEach(() => {
    render(<Title {...props} />);
  });

  it('calls back when clicked', () => {
    fireEvent.press(screen.getByTestId('back-button'));

    expect(props.handleBack).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
