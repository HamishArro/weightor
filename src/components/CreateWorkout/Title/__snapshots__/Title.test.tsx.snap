// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title tests matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#003844",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 8,
      "width": "100%",
    }
  }
  testID="title-container"
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 20,
      }
    }
  >
    title
  </Text>
  <Text
    allowFontScaling={false}
    onPress={
      [MockFunction] {
        "calls": [
          [],
        ],
        "results": [
          {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    selectable={false}
    style={
      [
        {
          "color": "white",
          "fontSize": 20,
        },
        undefined,
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="back-button"
  >
    î˜’
  </Text>
</View>
`;
